### PR TITLE
Moment map units match surface brightness+spectral units

### PR DESCRIPTION
**Pull request by bmorris3**
**Note**: This PR is in draft mode to begin discussion until the next release of specutils. Before this PR gets merged, we need a specutils release, and to bump the specutils minpin in this PR.

### Description

Recent Unit Conversion plugin updates led the units of the zeroth moment map to convert to the surface brightness unit on the y-axis of the spectrum-viewer. Previously, POs had decided on the default units of the zeroth moment map, which do not always match these "conventional"/"preferred" units. 

This PR restores that preferred behavior for the moment map, and simplifies the logic used to get there.

### Change log entry

- [ ] Is a change log needed? If yes, is it added to `CHANGES.rst`? If you want to avoid merge conflicts,
  list the proposed change log here for review and add to `CHANGES.rst` before merge. If no, maintainer
  should add a `no-changelog-entry-needed` label.

### Checklist for package maintainer(s)

This checklist is meant to remind the package maintainer(s) who will review this pull request of some common things to look for. This list is not exhaustive.

- [ ] Are two approvals required? Branch protection rule does not check for the second approval. If a second approval is not necessary, please apply the `trivial` label.
- [ ] Do the proposed changes actually accomplish desired goals? Also manually run the affected example notebooks, if necessary.
- [ ] Do the proposed changes follow the [STScI Style Guides](https://github.com/spacetelescope/style-guides)?
- [ ] Are tests added/updated as required? If so, do they follow the [STScI Style Guides](https://github.com/spacetelescope/style-guides)?
- [ ] Are docs added/updated as required? If so, do they follow the [STScI Style Guides](https://github.com/spacetelescope/style-guides)?
- [ ] Did the CI pass? If not, are the failures related?
- [ ] Is a milestone set? Set this to bugfix milestone if this is a bug fix and needs to be released ASAP; otherwise, set this to the next major release milestone. Bugfix milestone also needs an accompanying backport label.
- [ ] After merge, any internal documentations need updating (e.g., JIRA, Innerspace)? [🐱](https://jira.stsci.edu/browse/JDAT-4651)

### CODE REVIEW

File: jdaviz/conftest.py [hunk 1]
```diff
@@ -241,14 +241,23 @@ def spectrum1d_cube_larger():
 
 @pytest.fixture
 def spectrum1d_cube_largest():
+    return _spectrum1d_cube_largest_custom_unit(fluxunit=u.Jy)
+
+
+@pytest.fixture
+def spectrum1d_cube_largest_custom_unit():
+    return _spectrum1d_cube_largest_custom_unit
```
Comment (pllim) [inline jdaviz/conftest.py:249]: Missing function call here?

Comment (bmorris3) [reply jdaviz/conftest.py:249]: This is intentional. It let's us use the same function for two fixtures:

https://github.com/spacetelescope/jdaviz/blob/5c3855728fb03b752dfba3678813373a4b982a04/jdaviz/conftest.py#L242-L260

**Comment by kecnry** [comment]
is specutils 1.16 (released last week) what is needed here?

**Comment by pllim** [comment]
> what is needed here?

A serious rebase. Lots of conflicts.

**Comment by bmorris3** [comment]
This PR is now failing on tests related to fixes that @cshanahan1 is working in her latest ticket on surface brightness in units of pix^-2. Waiting for progress on [JDAT-4520](https://jira.stsci.edu/browse/JDAT-4520).

File: pyproject.toml [hunk 2]
```diff
@@ -23,7 +23,7 @@ dependencies = [
     "ipywidgets>=8.0.6",
     "voila>=0.4,<0.5",
     "pyyaml>=5.4.1",
-    "specutils>=1.15",
+    "specutils >= 0.16",
```
Comment (kecnry) [inline pyproject.toml:26]:
```suggestion
    "specutils >= 1.16",
```

**Pull request closed without merging**